overlord/hookstate/hooktest/handler.go

```diff
@@ -41,6 +45,10 @@ func NewMockHandler() *MockHandler {
 
 // Before satisfies hookstate.Handler.Before
 func (h *MockHandler) Before() error {
+	executing := atomic.AddInt32(&h.Executing, 1)
+	if executing != 1 {
+		panic(fmt.Sprintf("More than one handler executed: %d", executing))
```

Comment:
The logic here doesn't feel entirely sensible because we do allow handlers to execute concurrently as long as they're taking care of different snaps. Your test only passed because there was no second snap on it.

One might also argue that every execution should create a new handler, and that's what we tend to do, but then the logic is bogus for the opposite reason: if every hook has its own handler, this wouldn't catch the problem either.